hibernate-core/src/test/java/org/hibernate/test/onetoone/bidirectional/BiDirectionalOneToOneFetchTest.java

```diff
@@ -153,12 +154,40 @@ public void jdbcExecuteStatementStart() {
 		} );
 	}
 
+	@Test
+	@TestForIssue( jiraKey = "HHH-12885" )
+	public void testSelectInverseOneToOne() {
+		inTransaction( session -> {
+			EntityA a = new EntityA( 1L, new EntityB( 2L ) );
+
+			session.persist( a );
+			session.flush();
+			session.clear();
+
+			List<EntityA> list = session.createQuery(
+					"select a from EntityB b left join b.a a",
+					EntityA.class
+			).list();
+
+			List<Object[]> tupleList = session.createQuery(
+					"select b, b.id, a from EntityB b left join b.a a",
```

Comment:
Perhaps we should add 1 or 2 more cases with reordering the scalars a bit and gather some more confidence regarding the assumptions on the ordering of the arrays. What about `select b, b.id, a, a.id from EntityB b left join b.a a`?